packages/next/src/server/node-web-streams-helper.ts

```diff
@@ -215,6 +215,13 @@ function createHeadInsertionTransformStream(
         })
       }
     },
+    async flush(controller) {
+      // Check before closing if there's anything remaining to insert.
+      const insertion = await insert()
+      if (insertion) {
+        controller.enqueue(encodeText(insertion))
+      }
+    },
```

Comment:
Can you articulate what changed in my PR that made this omission start to become an issue? It seems like this should have been a problem already

Reply:
I believe it has been a problem already for a long time, and it's flaky - even with this PR I can still pass the test. So I believe there's a race condition somewhere and some new React changes make that easier to appear.

Reply:
right but isn't this production code? so you just fixed a real issue